climada/hazard/low_flow.py

```diff
@@ -357,11 +417,11 @@ def _df_clustering(data, cluster_vars, res_data, clus_thres_xy,
             cluster_id = cluster_id[x_y_cpy]
             data_iter[clus_id_var].values[temp] = cluster_id + data_iter[clus_id_var].max() + 1
 
-        data[clus_id_var].values[data['iter_ev'].values] = data_iter[clus_id_var].values
-        return data
+        data_df[clus_id_var].values[data_df['iter_ev'].values] = data_iter[clus_id_var].values
+        return data_df
 
     def filter_events(self, min_intensity=1, min_number_cells=1):
```

Comment:
Why use haz_tmp.date to select events to filter out? What if someone is interested in events in November of the year 5476? Why not create an array of True with the right dimension, set them to false if you want to filter them out and select the events by using the array of bools to index event_names.

Reply:
`hazard.select()` has limited input parameters that are possible, but I found a better way to filter using `self.orig` instead of `self.date`.